test/src/test/java/org/corfudb/integration/LogSizeQuotaIT.java

```diff
@@ -99,4 +99,58 @@ public void testLogSizeQuota() throws Exception {
         assertThat(map2.get("k4")).isEqualTo(map.get("k4"));
         shutdownCorfuServer(server_1);
     }
+
+    @Test
+    public void testLogSizeQuotaViaRuntime() throws Exception {
+
+        final int numWrites = 1000;
+        final int payloadSize = 100;
+
+        Process server_1 = new CorfuServerRunner()
+                .setHost(DEFAULT_HOST)
+                .setPort(DEFAULT_PORT)
+                .setSingle(true)
+                .setLogPath(getCorfuServerLogPath(DEFAULT_HOST, DEFAULT_PORT))
+                .setLogSizeQuota(Long.toString(numWrites * payloadSize))
+                .runServer();
+
+        byte[] payload = new byte[payloadSize];
+
+        // Configure a client with a max write limit
+        CorfuRuntime.CorfuRuntimeParameters params = CorfuRuntime.CorfuRuntimeParameters
+                .builder()
+                .writePriority(Priority.HIGH)
+                .build();
+
+        CorfuRuntime rt = CorfuRuntime.fromParameters(params);
+        rt.parseConfigurationString(DEFAULT_ENDPOINT);
+        rt.connect();
+
+        Map<String, String> map = rt.getObjectsView()
+                .build()
+                .setStreamName("s1")
+                .setType(CorfuTable.class)
+                .open();
+
+        // Create a map
+        map.put("k1", "v1");
+        map.put("k2", "v2");
+
+        // Fill the log with data till the quota is reached
+        IStreamView sv = rt.getStreamsView().get(UUID.randomUUID());
+
+        boolean quotaReached = false;
+        for (int x = 0; x < numWrites; x++) {
```

Comment:
Since it gets right to the limit of the quota, should you afterwards also exceed the quota to make the test a bit more complete?

Reply:
it is part of testLogSizeQuota() but good point let me make that change. thanks